index.bs

```diff
@@ -292,16 +295,33 @@ the PrivacyCG.
 
 * TODO: Define how bounce tracking information is stored; e.g. sites,
         timestamps, etc.
-* TODO: Define how site interaction information is stored.
 * TODO: Define a recurring global timer to run the analyze and delete algorithm.
 
+The user agent contains a <dfn>user activation map</dfn> which is a [=map=] of
+[=sites=] to [=moments=].
+
 <h3 id="bounce-tracking-mitigations-algorithms">Algorithms</h3>
 
 * TODO: Define the steps necessary to detect and store a "bounce".
-* TODO: Define the steps necessary to store user interactions.
 * TODO: Define the steps to analyze information in the data model and delete
         appropriate sites.
 
+<h4 id="bounce-tracking-mitigations-activation-monkey-patch">User Activation Monkey Patch</h4>
+
+Append the following steps to the [=activation notification=] steps in the
+[[HTML#user-activation-processing-model|user activation processing model]].
+
+1. Assert |document| is not null.
+1. Let |context| be |document|'s [=Document/browsing context=].
+1. If |context| is null, then abort these steps.
+1. Let |topLevelDocument| be |context|'s [=top-level browsing context=]'s
+    [=navigable/active document=].
+1. Let |origin| be |topLevelDocument|'s [=Document/origin=].
```

Comment:
Do you maybe want the |document|'s [=relevant settings object=]'s [[=environment/top level origin=]](https://html.spec.whatwg.org/#concept-environment-top-level-origin)?
It's less clear what that means, but maybe we can define it now.

Reply:
As mentioned separately I actually spec'd the wrong thing here.  I want to avoid recording anything if the activation was not at the top-level.  I'll rework the PR.